src/Application/ZoomApplication.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * This file is part of the Secret Santa project.
+ *
+ * (c) JoliCode <coucou@jolicode.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JoliCode\SecretSanta\Application;
+
+use JoliCode\SecretSanta\Model\SecretSanta;
+use JoliCode\SecretSanta\Model\User;
+use JoliCode\SecretSanta\Zoom\MessageSender;
+use JoliCode\SecretSanta\Zoom\UserExtractor;
+use League\OAuth2\Client\Token\AccessToken;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+class ZoomApplication implements ApplicationInterface
+{
+    const APPLICATION_CODE = 'zoom';
+
+    const SESSION_KEY_STATE = 'santa.zoom.state';
+
+    private const SESSION_KEY_TOKEN = 'santa.zoom.token';
+    private const SESSION_KEY_BOT_TOKEN = 'santa.zoom.bot_token';
+    private const SESSION_KEY_ADMIN = 'santa.zoom.admin';
+    private const SESSION_KEY_ACCOUNT_ID = 'santa.zoom.account_id';
+
+    private $requestStack;
+    private $userExtractor;
+    private $messageSender;
+
+    public function __construct(RequestStack $requestStack, UserExtractor $userExtractor, MessageSender $messageSender)
+    {
+        $this->requestStack = $requestStack;
+        $this->userExtractor = $userExtractor;
+        $this->messageSender = $messageSender;
+    }
+
+    public function getCode(): string
+    {
+        return self::APPLICATION_CODE;
+    }
+
+    public function isAuthenticated(): bool
+    {
+        try {
+            $this->getToken();
+
+            return true;
+        } catch (\LogicException $e) {
+            return false;
+        }
+    }
+
+    public function getAuthenticationRoute(): string
+    {
+        return 'zoom_authenticate';
+    }
+
+    public function getOrganization(): string
+    {
+        return $this->getAccountId(); // todo: there is no team name :/
```

Comment:
What about the todo?

Reply:
Already reported in the issue that we cannot do better at the moment. It a non blocker.